tests/integration/cloudformation/resources/test_lambda.py

```diff
@@ -795,3 +796,127 @@ def wait_logs():
 
         with pytest.raises(lambda_client.exceptions.ResourceNotFoundException):
             lambda_client.get_event_source_mapping(UUID=esm_id)
+
+
+class TestCfnLambdaDestinations:
+    """
+    generic cases
+    1. verify payload
+
+    - [ ] SNS destination success
+    - [ ] SNS destination failure
+    - [ ] SQS destination success
+    - [ ] SQS destination failure
+    - [ ] Lambda destination success
+    - [ ] Lambda destination failure
+    - [ ] EventBridge destination success
+    - [ ] EventBridge destination failure
+
+    meta cases
+    * test max event age
+    * test retry count
+    * qualifier issues
+    * reserved concurrency set to 0 => should immediately go to failure destination / dlq
+    * combination with DLQ
+    * test with a very long queue (reserved concurrency 1, high function duration, low max event age)
+
+    edge cases
+    - [ ] Chaining async lambdas
+
+    doc:
+    "If the function doesn't have enough concurrency available to process all events, additional requests are throttled.
+    For throttling errors (429) and system errors (500-series), Lambda returns the event to the queue and attempts to run the function again for up to 6 hours.
+    The retry interval increases exponentially from 1 second after the first attempt to a maximum of 5 minutes.
+    If the queue contains many entries, Lambda increases the retry interval and reduces the rate at which it reads events from the queue."
+
+    """
+
+    @pytest.mark.skip(reason="not supported atm and test needs further work")
+    @pytest.mark.parametrize(
+        ["on_success", "on_failure"],
+        [
+            ("sqs", "sqs"),
+            # ("sns", "sns"),
+            # ("lambda", "lambda"),
+            # ("eventbridge", "eventbridge")
+        ],
+    )
+    @pytest.mark.aws_validated
+    def test_generic_destination_routing(
+        self, lambda_client, logs_client, deploy_cfn_template, cfn_client, on_success, on_failure
+    ):
+        """
+        This fairly simple template lets us choose between the 4 different destinations for both OnSuccess as well as OnSuccess.
```

Comment:
Both OnSuccess and OnFailure, right?
```suggestion
        This fairly simple template lets us choose between the 4 different destinations for both OnSuccess as well as OnFailure.
```